prow.sh

```diff
@@ -859,14 +861,17 @@ install_snapshot_controller() {
       # shellcheck disable=SC2001
       modified="$(echo "$yaml" | sed -e "s;image: .*/\([^/:]*\):.*;image: ${CSI_PROW_DRIVER_CANARY_REGISTRY}/\1:canary;")"
       diff <(echo "$yaml") <(echo "$modified")
-      if ! echo "$modified" | kubectl apply -f -; then
-          echo "modified version of $SNAPSHOT_CONTROLLER_YAML:"
-          echo "$modified"
-          exit 1
-      fi
-  else
-      echo "kubectl apply -f $SNAPSHOT_CONTROLLER_YAML"
-      kubectl apply -f "$SNAPSHOT_CONTROLLER_YAML"
+      snapshot_yaml="$modified"
+  elif volume_mode_conversion; then
```

Comment:
Yup, that's the assumption right now. The e2e tests are going to be added to the provisioner repo.

Reply:
Did some refactoring to enable the feature irrespective of how the script is being called

Reply:
Then it makes more sense to me to change the way how the snapshotter gets installed only in that repo. It keeps csi-release-tools simpler and, more importantly, you can make all changes for the E2E tests in one place.

You can do something like this in external-provisioner/.prow.sh:

```
. release-tools/prow.sh

volume_mode_conversion () {
  [ "${VOLUME_MODE_CONVERSION_TESTS}" == "true" ]
}

if volume_mode_conversion; then
    install_snapshot_controller() {
        # install controller differently
    }
}

main
```

Reply:
@pohly your suggestion makes sense. Before I make the change, can you take a look at the latest commit? 
I've added a function to the csi-release-tools script enable snapshot-controller feature flags (can be updated with more features in the future). It modifies the yaml before attempting to install the snapshot-controller, so this flag can be passed in irrespective of whether it is being called within snapshotter/with canary images/ otherwise. Let me know what you think is the better approach. 

I think we've been running into a bit of a race condition with our comments :)